test/02/t0260a.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/sh
+#
+# srecord - Manipulate EPROM load files
+# Copyright (C) 2014 Markus Heidelberg
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or (at
+# your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+
+TEST_SUBJECT="random fill near end of address space"
+. test_prelude
+
+cat > test.ok << 'fubar'
+Format: Motorola S-Record
+Header: "HDR"
+Execution Start Address: 00000000
+Data:   00000000 - 00000000
+        FFFFFFFE - FFFFFFFE
+fubar
+if test $? -ne 0; then no_result; fi
+
+srec_cat -generate 0 1 -constant 0 -random-fill 0xFFFFFFFE 0xFFFFFFFF \
+        -o test.srec -header HDR -start-addr 0
+if test $? -ne 0; then fail; fi
+
+srec_info test.srec > test.out
+if test $? -ne 0; then fail; fi
+
+diff test.ok test.out
+if test $? -ne 0; then fail; fi
+
+# use the last address
+# -> buggy srec_info output is a different issue
+
+cat > test.ok << 'fubar'
+Format: Motorola S-Record
+Header: "HDR"
+Execution Start Address: 00000000
+Data:   0000 - 0000
+        FFFFFFFF - FFFFFFFFFFFFFFFF
+fubar
```

Comment:
srec_info also seems to have some problems with the last address. I just noticed it when adding the second test, but didn't dig into it.
(I use srec_info here for comparison because the output with the random-fill filter is not constant)

Reply:
Ok, that is odd. I hope it isn't something I introduced with the verbose additions to srec_info. Definitely keen for one of us to fix it though. Would you mind creating an issue to capture an example?
Using srec_info to test the output is a neat idea! I had considered adding a RNG seed option for random to help solve a similar problem but it was a useless feature otherwise.

Reply:
Issue #10 created.
The second data line above (FFFFFFFFFFFFFFFF) is indeed a regression, the first data line (0000 instead of 00000000) was the case in version 1.64 already.